src/OrleansTestKit/Storage/TestStorage.cs

```diff
@@ -14,6 +14,8 @@ internal sealed class TestStorage<TState> :
 
         public string Etag => throw new System.NotImplementedException();
 
+        public virtual bool RecordExists => throw new NotImplementedException();
```

Comment:
Can I suggest `public bool RecordExists => Stats.Clears >= Stats.Writes;`

Reply:
I'll dig into the stats calls. However, iirc the writes counter is not reset after a call to `ClearStateAsync`. As such, it's possible that a test has `Stats.Writes > Stats.Clears`, but the _last call_ was to `CleearStateAsync`, in which case `RecordExists` should still return `false`.

Reply:
ahh - fair enough.  So then set the flag to true on write and false on clear.